README.md

```diff
@@ -95,6 +95,55 @@ However, here are some rules that we'd like to emphasize since they are not visi
+|Expected Outcome         | A benchmarking framework that mimics JMH APIs and produces reliable results for Scala Native and Scala.js. Demonstration benchmarks showcasing usage. |
+|Prerequisites            | Good understanding of benchmarking principles, Scala Native, Scala.js, and Scala compiler plugins. Some knowledge of JMH is beneficial. |
+|Expected Difficulty      | Medium |
+|Expected Time Commitment | Medium project - 175 hours |
+|Mentor                   | Wojciech Mazur (GitHub: [@WojciechMazur](https://github.com/Wojciech Mazur), Email: [wmazur@virtuslab.com](wmazur@virtuslab.com))|
+|Co-mentor                | TODO |
+
+### Scala Native Projects
+
+| Title                   | WebAssembly Backend for Scala Native |
+|-------------------------|--------------------------------------|
+|Link to Project          | [Scala Native](https://scala-native.org/) |
+|Brief Description        | This project aims to introduce WebAssembly (Wasm) as a new backend for Scala Native, allowing Scala Native programs to run efficiently in WebAssembly-supported environments, such serverless runtimes (via WASI) or browsers. The goal is to design an alternative to current LLVM-based backend allowing to emit Wasm directlly |
+|Expected Outcome         | A functional WebAssembly backend capable of compiling Scala Native programs to Wasm, with basic runtime support and test cases demonstrating correctness. |
+|Prerequisites            | Strong knowledge of Scala and compilers. Familiarity with LLVM, WebAssembly, and Scala Native’s internals is a plus. |
+|Expected Difficulty      | Hard |
+|Expected Time Commitment | Large project - 350 hours |
+|Mentor                   |Wojciech Mazur (GitHub: [@WojciechMazur](https://github.com/Wojciech Mazur), Email: [wmazur@virtuslab.com](wmazur@virtuslab.com))|
+|Co-mentor                | TODO |
+
+| Title                   | Incremental Optimization of Scala Native IR |
+|-------------------------|-----------------------------------------------|
+|Link to Project          | [Scala Native](https://scala-native.org/) |
+|Brief Description        | This project aims to implement incremental optimization for Scala Native’s Intermediate Representation (IR), improving compilation speed by reusing results from previous compilation runs instead of optimizing the entire program from scratch. |
+|Expected Outcome         | A working prototype of an incremental optimization pipeline for Scala Native IR, with measurable speedups over full optimizations. |
+|Prerequisites            | Strong understanding of compilers, Scala Native IR, and optimization techniques. Knowledge of LLVM and incremental compilation strategies is a plus. |
+|Expected Difficulty      | Hard |
+|Expected Time Commitment | Large project - 350 hours |
+|Mentor                   |Wojciech Mazur (GitHub: [@WojciechMazur](https://github.com/Wojciech Mazur), Email: [wmazur@virtuslab.com](wmazur@virtuslab.com))|
+|Co-mentor                | TODO |
```

Comment:
@sjrd Would you be willing to co-mentor based on your experience with designing Scala.js incremental optimiser?

Reply:
I can do that, yes.